File: PoketDex/app/src/main/java/pt/rfsfernandes/custom/utils/UtilsClass.java
package pt.rfsfernandes.custom.utils;

import android.content.Context;

import pt.rfsfernandes.R;

public class UtilsClass {

  /**
   * A method to turn pokemon names with hifen (-) into camel cased name
   *
   * @param word to camel case
   * @return String in camel case
   */
  public static String toCamelCase(String word) {
    String[] s = word.split("-");
    StringBuilder newWord = new StringBuilder();

    for (String s1 : s) {
      char[] chars = s1.toCharArray();
      for (int i = 0; i < chars.length; i++) {
        if (i == 0)
          newWord.append(String.valueOf(chars[i]).toUpperCase());
        else
          newWord.append(String.valueOf(chars[i]).toLowerCase());
      }
      newWord.append(" ");
    }

    newWord.trimToSize();

    return newWord.toString();
  }

  /**
   * Returns the color matching the name
   *
   * @param typeName Pokemon Type
   * @return Color of corresponding pokemon type
   */
  public static int returnColorId(Context context, String typeName) {
    int colorId = 0;

    switch (typeName.toLowerCase()) {
      case "bug":
        colorId = context.getResources().getColor(R.color.bug, null);
        break;
      case "dark":
        colorId = context.getResources().getColor(R.color.dark, null);
        break;
      case "dragon":
        colorId = context.getResources().getColor(R.color.dragon, null);
        break;
      case "electric":
        colorId = context.getResources().getColor(R.color.electric, null);
        break;
      case "fairy":
        colorId = context.getResources().getColor(R.color.fairy, null);
        break;
      case "fighting":
        colorId = context.getResources().getColor(R.color.fighting, null);
        break;
      case "fire":
        colorId = context.getResources().getColor(R.color.fire, null);
        break;
      case "flying":
        colorId = context.getResources().getColor(R.color.flying, null);
        break;
      case "ghost":
        colorId = context.getResources().getColor(R.color.ghost, null);
        break;
      case "grass":
        colorId = context.getResources().getColor(R.color.grass, null);
        break;
      case "ground":
        colorId = context.getResources().getColor(R.color.ground, null);
        break;
      case "ice":
        colorId = context.getResources().getColor(R.color.ice, null);
        break;
      case "normal":
        colorId = context.getResources().getColor(R.color.normal, null);
        break;
      case "poison":
        colorId = context.getResources().getColor(R.color.poison, null);
        break;
      case "psychic":
        colorId = context.getResources().getColor(R.color.psychic, null);
        break;
      case "rock":
        colorId = context.getResources().getColor(R.color.rock, null);
        break;
      case "steel":
        colorId = context.getResources().getColor(R.color.steel, null);
        break;
      case "water":
        colorId = context.getResources().getColor(R.color.water, null);
        break;

    }

    return colorId;
  }

}

File: PocketDex/app/src/main/java/pt/rfsfernandes/pocketdex/custom/utils/UtilsClass.kt
package pt.rfsfernandes.pocketdex.custom.utils

import android.content.Context
import pt.rfsfernandes.pocketdex.R
import java.util.*

object UtilsClass {
    /**
     * A method to turn pokemon names with hifen (-) into camel cased name
     *
     * @param word to camel case
     * @return String in camel case
     */
    fun toCamelCase(word: String): String {
        val s = word.split("-").toTypedArray()
        val newWord = StringBuilder()
        for (s1 in s) {
            val chars = s1.toCharArray()
            for (i in chars.indices) {
                if (i == 0) newWord.append(
                    chars[i].toString().uppercase(Locale.getDefault())
                ) else newWord.append(chars[i].toString().lowercase(Locale.getDefault()))
            }
            newWord.append(" ")
        }
        newWord.trimToSize()
        return newWord.toString()
    }

    /**
     * Returns the color matching the name
     *
     * @param typeName Pokemon Type
     * @return Color of corresponding pokemon type
     */
    fun returnColorId(context: Context, typeName: String): Int {
        var colorId = 0
        when (typeName.lowercase(Locale.getDefault())) {
            "bug" -> colorId = context.resources.getColor(R.color.bug, null)
            "dark" -> colorId = context.resources.getColor(R.color.dark, null)
            "dragon" -> colorId = context.resources.getColor(R.color.dragon, null)
            "electric" -> colorId = context.resources.getColor(R.color.electric, null)
            "fairy" -> colorId = context.resources.getColor(R.color.fairy, null)
            "fighting" -> colorId = context.resources.getColor(R.color.fighting, null)
            "fire" -> colorId = context.resources.getColor(R.color.fire, null)
            "flying" -> colorId = context.resources.getColor(R.color.flying, null)
            "ghost" -> colorId = context.resources.getColor(R.color.ghost, null)
            "grass" -> colorId = context.resources.getColor(R.color.grass, null)
            "ground" -> colorId = context.resources.getColor(R.color.ground, null)
            "ice" -> colorId = context.resources.getColor(R.color.ice, null)
            "normal" -> colorId = context.resources.getColor(R.color.normal, null)
            "poison" -> colorId = context.resources.getColor(R.color.poison, null)
            "psychic" -> colorId = context.resources.getColor(R.color.psychic, null)
            "rock" -> colorId = context.resources.getColor(R.color.rock, null)
            "steel" -> colorId = context.resources.getColor(R.color.steel, null)
            "water" -> colorId = context.resources.getColor(R.color.water, null)
        }
        return colorId
    }
}